src/main/java/com/jagrosh/giveawaybot/commands/SettingsCommand.java

```diff
@@ -47,13 +56,125 @@ public SettingsCommand(Bot bot)
+        {
+            resetBlock(event, level);
+            return;
+        }
+
+        if (!level.canSetEmoji())
+        {
+            event.replyError("This server must have a premium level to set a custom emoji!");
+            return;
+        }
+
+        if (args[1].length() > 60)
+        {
+            event.replyWarning("It seems like you entered multiple emojis. Please enter only one valid emoji.");
+            return;
+        }
+
+        String extracted;
+
+        if (args[1].matches(EMOTE_REGEX))
+        {
+            extracted = args[1].substring(1, args[1].length() - 1);
+        }
+        else if (EmojiManager.isEmoji(args[1]))
+        {
+            extracted = args[1];
+        }
+        else
+        {
+            event.replyError("The provided emoji seems to be invalid.");
+            return;
+        }
+
+        final String finalExtracted = extracted; // because of lambda expression
+        event.getMessage().addReaction(finalExtracted)
+                .map((success) -> {
+                    bot.getDatabase().settings.updateEmoji(event.getGuild(), finalExtracted);
+                    event.replySuccess("Successfully set " + args[1] + " as the new servers reaction emoji.");
+                    event.getMessage().removeReaction(finalExtracted, event.getSelfUser()).queue();
+                    return null;
+                })
+                .onErrorMap((error) -> {
+                    event.replyWarning("The provided emoji is not accessible for me. Please use a different one.");
+                    return null;
+                }).queue();
+    }
+
+    private void resetBlock(CommandEvent event, PremiumLevel level) {
+        // check for running giveaways
+        List<Giveaway> list = level.perChannelMaxGiveaways
+                ? bot.getDatabase().giveaways.getGiveaways(event.getTextChannel())
+                : bot.getDatabase().giveaways.getGiveaways(event.getGuild());
+        if(list == null)
+        {
+            event.replyError("An error occurred when trying to change settings.");
+            return;
+        }
+        else if(!list.isEmpty())
+        {
+            event.replyError("There are already " + level.maxGiveaways + " giveaways running in this "
+                    + (level.perChannelMaxGiveaways ? "channel" : "server") + "!");
+            return;
+        }
+        if (bot.getDatabase().settings.getSettings(event.getGuild().getIdLong()).getEmojiDisplay().equals(Constants.TADA))
+            return; // might be redundant check, will remove if desired
+        bot.getDatabase().settings.updateEmoji(event.getGuild(), null);
+    }
+
+    private void defaultBlock(CommandEvent event, EmbedBuilder eb, GuildSettings settings, PremiumLevel level)
+    {
+        String displayedEmoji = (settings.getEmojiDisplay().contains(":")) ? ("<" + settings.getEmojiDisplay() + ">") : settings.getEmojiDisplay();
```

Comment:
This shouldn't be necessary; the `getEmojiDisplay()` method should already make an in-discord renderable version of the emoji, and there needs to be a new `getEmojiRaw()` method that gives the underlying variant that is saved and can be reacted with.

Note that this will require changing the existing `getEmojiDisplay()` since right now it just does what `getEmojiRaw()` will need to do.